docs/howto/pipeline_testing.md

```diff
@@ -0,0 +1,183 @@
+# HOWTO: Writing pipeline tests for a package
+
+## Introduction
+
+Elastic Packages are comprised of data streams. A pipeline test exercises ingest pipelines defined for a package's data stream.
+
+## Conceptual process
+
+Conceptually, running a pipeline test involves the following steps:
+
+1. Deploy the Elasticsearch instance (part of the Elastic Stack). This step takes time so it should typically be done once as a pre-requisite to running pipeline tests on multiple data streams.
+1. Upload ingest pipelines to be tested.
+1. Use [Simulate API](https://www.elastic.co/guide/en/elasticsearch/reference/master/simulate-pipeline-api.html) to process logs/metrics with the ingest pipeline.
+1. Compare generated results with expected ones.
+
+## Limitations
+
+At the moment pipeline tests have limitations. The main ones are:
+* Results of simulation can be different from real documents. They depend on the input data which may not have all fields defined.
+* There might be integrations which transform data mostly using Beats processors instead of ingest pipelines. In such cases ingest pipelines are rather plain.
```

Comment:
> In such cases ingest pipelines are rather plain.

Interesting. If a data stream is using **only** Beats processors, I wonder if it even ends up creating an ingest pipeline in Elasticsearch. 🤔

Reply:
The ingest pipeline is not mandatory. You can take a look at the `log` integration, where the pipeline is not defined:

https://github.com/elastic/integrations/tree/master/packages/log/data_stream/log